### DISCUSSION TITLE
How to specify a compile_commands.json path for clangd?

### DISCUSSION THREAD

**Discussion by missdeer** [zed-industries/zed#6480] [Feature Requests]
Any documentation about clangd settings?

**Comment by photex** [comment]
I also wondered a bit about this. I figured it's not quite available so I just use 'ln -s' to bring it to the repo root

**Reply by photex** [reply]
Hrm... "suggested answer"? Wasn't my intention! Just commenting. :D

**Reply by missdeer** [reply]
> I also wondered a bit about this. I figured it's not quite available so I just use 'ln -s' to bring it to the repo root

It works for me. Thanks a lot.

**Comment by zambetti** [comment]
I've found that I can more-or-less hide `compile_commands.json` in a `.clang` folder in my project directory `./.clang/compile_commands.json` by using the configuration file `.clangd` (also in my project directory) as such...

```
CompileFlags:
  CompilationDatabase: "./.clang"
```

I would also like to put the clangd config file in the `.clang` folder as `.clang/configuration.json` so my project folder it tidy, but to do this, I'd need to pass `--config-file` to clangd somehow. This seems like it should be possible in Zed's `settings.json` "lsp" node, but I can't seem to get it working.

I tried variants like this (below), but there's no documentation yet about how to configure clangd in settings.json

```
    "enable_language_server": true,
    "lsp":
    {
        "clangd": // None of this works
        {
            "arguments":
            [
                "--config-file=/.clang/configuration.yaml",
                "--compile-commands-dir=./.clang/"
            ]
        }
    },
   ```

Does anyone know if passing args to clangd is possible via Zed's settings.json, and if so, how to do it?

Related issue #4295

**Comment by UrielCh** [comment]
after some test:
the local .zed/settings.json looks to be ignore

the content of ~/.config/zed/settings.json looks to impact the language server.

using `zed --foreground` help debugging.

**Reply by ngn999** [reply]
As of 2025-04-29, `.zed/settings.json` is working.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links

**Comment by peterxcli** [comment]
This works:
```json
{
  "lsp": {
    "clangd": {
      "binary": {
        "arguments": ["--compile-commands-dir=./build/release"]
      }
    }
  }
}
```

**Reply by azizka85** [reply]
Yeah, it works! Thank you.
